### PR TITLE
Refactor `try rescue` statements to keep stacktrace

### LINKED ISSUE #6762: Refactor rescue statements, add __STACKTRACE__
Now we get such error:
```
2023-01-23T09:51:07.843 application=indexer fetcher=internal_transaction count=15 error_count=15 [error] failed to fetch internal transactions for blocks: %FunctionClauseError{args: nil, arity: 4, clauses: nil, function: :parse_call_tracer_calls, kind: nil, module: EthereumJSONRPC.Geth}
```
It has no function arguments, hard to debug.
Including `__STACKTRACE__` or eliminating `rescue` would make logs more informative

### PR DESCRIPTION
**Pull request by sl1depengwyn**
Closes #6762 

## Changelog

Example error before:
```
2023-01-30T20:20:44.907 application=indexer fetcher=internal_transaction count=10 error_count=10 [error] failed to fetch internal transactions for blocks: %FunctionClauseError{module: EthereumJSONRPC.Geth, function: :parse_call_tracer_calls, arity: 4, kind: nil, args: nil, clauses: nil}
```
after:
```
2023-01-30T20:21:39.989 application=indexer fetcher=internal_transaction count=10 error_count=10 [error] failed to fetch internal transactions for blocks: ** (FunctionClauseError) no function clause matching in EthereumJSONRPC.Geth.parse_call_tracer_calls/4
    (ethereum_jsonrpc 5.0.0) lib/ethereum_jsonrpc/geth.ex:233: EthereumJSONRPC.Geth.parse_call_tracer_calls({%{"from" => "0x0a70c7294f9deffe389e437347ba1b6de531d16c", "gas" => "0xeef6c", "gasUsed" => "0x48e8d", "input" => "0xa0712d680000000000000000000000000000000000000000000000000000000000000002", "output" => "0x", "to" => "0xd9106423df3a233a5033a3783e5ccac01647fc6c", "type" => "CALL", "value" => "0x0"}, 0}, [], [], false)
    (ethereum_jsonrpc 5.0.0) lib/ethereum_jsonrpc/geth.ex:227: EthereumJSONRPC.Geth.prepare_calls/1
    (ethereum_jsonrpc 5.0.0) lib/ethereum_jsonrpc/geth.ex:178: EthereumJSONRPC.Geth.debug_trace_transaction_response_to_internal_transactions_params/2
    (elixir 1.14.3) lib/enum.ex:1658: Enum."-map/2-lists^map/1-0-"/2
    (ethereum_jsonrpc 5.0.0) lib/ethereum_jsonrpc/geth.ex:129: EthereumJSONRPC.Geth.debug_trace_transaction_responses_to_internal_transactions_params/3
    (indexer 5.0.0) lib/indexer/fetcher/internal_transaction.ex:174: anonymous fn/3 in Indexer.Fetcher.InternalTransaction.fetch_block_internal_transactions_by_transactions/2
    (elixir 1.14.3) lib/enum.ex:2468: Enum."-reduce/3-lists^foldl/2-0-"/3
    (indexer 5.0.0) lib/indexer/fetcher/internal_transaction.ex:115: Indexer.Fetcher.InternalTransaction.run/2
    (elixir 1.14.3) lib/task/supervised.ex:89: Task.Supervised.invoke_mfa/2
    (elixir 1.14.3) lib/task/supervised.ex:34: Task.Supervised.reply/4
    (stdlib 4.2) proc_lib.erl:240: :proc_lib.init_p_do_apply/3

```

## Checklist for your Pull Request (PR)

  - [x] I added an entry to `CHANGELOG.md` with this PR
  - [ ] If I added new functionality, I added tests covering it.
  - [ ] If I fixed a bug, I added a regression test to prevent the bug from silently reappearing again.
  - [ ] I checked whether I should update the docs and did so by submitting a PR to https://github.com/blockscout/docs
  - [ ] If I added/changed/removed ENV var, I submitted a PR to https://github.com/blockscout/docs to update the list of env vars at https://github.com/blockscout/docs/blob/master/for-developers/information-and-settings/env-variables.md and I updated the version to `master` in the Version column. Changes will be reflected in this table: https://docs.blockscout.com/for-developers/information-and-settings/env-variables. 
  - [ ] If I add new indices into DB, I checked, that they are not redundant with PGHero or other tools

### CODE REVIEW

File: apps/indexer/lib/indexer/fetcher/internal_transaction.ex [hunk 1]
```diff
@@ -101,29 +101,37 @@ defmodule Indexer.Fetcher.InternalTransaction do
...
     |> Keyword.fetch!(:variant)
     |> case do
-      EthereumJSONRPC.Nethermind ->
-        EthereumJSONRPC.fetch_block_internal_transactions(filtered_unique_numbers, json_rpc_named_arguments)
-
-      EthereumJSONRPC.Erigon ->
-        EthereumJSONRPC.fetch_block_internal_transactions(filtered_unique_numbers, json_rpc_named_arguments)
-
-      EthereumJSONRPC.Besu ->
+      variant when variant in [EthereumJSONRPC.Nethermind, EthereumJSONRPC.Erigon, EthereumJSONRPC.Besu] ->
```
Comment (sl1depengwyn) [inline apps/indexer/lib/indexer/fetcher/internal_transaction.ex:104]: To fix appeared `Function is too complex` credo error

**Review by vbaranov — APPROVED** [review]

File: apps/ethereum_jsonrpc/lib/ethereum_jsonrpc/geth.ex [hunk 2]
```diff
@@ -231,10 +231,10 @@ defmodule EthereumJSONRPC.Geth do
 
   defp parse_call_tracer_calls(calls, acc, trace_address, inner? \\ true)
   defp parse_call_tracer_calls([], acc, _trace_address, _inner?), do: acc
-  defp parse_call_tracer_calls({%{"type" => 0}, _}, acc, _trace_address, _inner?), do: acc
+  defp parse_call_tracer_calls({%{"type" => 0, "asd" => 123}, _}, acc, _trace_address, _inner?), do: acc
 
   defp parse_call_tracer_calls(
-         {%{"type" => type, "from" => from} = call, index},
+         {%{"type" => type, "from" => from, "asd" => 123} = call, index},
```
Comment (nikitosing) [inline apps/ethereum_jsonrpc/lib/ethereum_jsonrpc/geth.ex:237]: what is the reason for addind such key here and above?

Comment (sl1depengwyn) [reply apps/ethereum_jsonrpc/lib/ethereum_jsonrpc/geth.ex:237]: Sorry, debug artifacts

File: apps/explorer/lib/explorer/token/instance_metadata_retriever.ex [hunk 3]
```diff
@@ -161,7 +161,7 @@ defmodule Explorer.Token.InstanceMetadataRetriever do
     fetch_json(%{@uri => {:ok, [decoded_json]}}, hex_token_id)
   rescue
     e ->
-      Logger.debug(["Unknown metadata format #{inspect(json)}. error #{inspect(e)}"],
+      Logger.debug(["Unknown metadata format #{inspect(json)}. error #{inspect(e)}", __STACKTRACE__],
```
Comment (nikitosing) [inline apps/explorer/lib/explorer/token/instance_metadata_retriever.ex:164]: Why did you leave `inspect(error)` in this file?

**Pull request merged**